lib/i18n/js.rb

```diff
@@ -71,11 +71,12 @@ def self.merge_with_fallbacks!(result)
     end
 
     def self.filtered_translations
-      {}.tap do |result|
+      translations = {}.tap do |result|
         translation_segments.each do |segment|
           Utils.deep_merge!(result, segment.translations)
         end
       end
+      Utils.deep_key_sort(translations) if I18n::JS.sort_translation_keys?
```

Comment:
This makes `filtered_translations` to return `nil` when `I18n::JS.sort_translation_keys?` is `false`